lib/liquid/context.rb

```diff
@@ -27,7 +27,7 @@ def initialize(environments = {}, outer_scope = {}, registers = {}, rethrow_erro
       @this_stack_used = false
 
       if rethrow_errors
-        self.exception_handler = ->(e) { raise }
+        self.exception_handler = ->(e) { raise e }
```

Comment:
I would argue that with `strict_filters` enabled you should not be able to reference an undefined filter. Unlike variables (which might not be defined yet since they are controlled by the user), filters are static and if undefined must be a typo.